### ISSUE TITLE
publish application API change

### ISSUE THREAD

**Issue by ajblum** [ryran/ravshello#67]
Ravello will be changing the API slightly within the next 5 months.  This change will affect the way ravshello publishes applications.  It will also affect the python SDK which will also be updated.

Here are the details of the change

Publish Application:
The call stays almost the same but there is no more need to choose the cloud (Amazon/Google)
- New

```
POST https://cloud.ravellosystems.com/api/v1/applications/34045955/publish
{ "preferredRegion": "us-central-1", "optimizationLevel": "PERFORMANCE_OPTIMIZED", "startAllVms": "true" }
```

-Old

```
POST https://cloud.ravellosystems.com/api/v1/applications/34045955/publish
{ "preferredCloud": "AMAZON", "preferredRegion": "Virginia", "optimizationLevel": "PERFORMANCE_OPTIMIZED", "startAllVms": "true" }
```

In addition, the region names will change, you can see the list of available regions on your account using this call:

```
GET https://cloud.ravellosystems.com/api/v1/applications/34045/publishLocations
```

Create Elastic IP:
Only change here is the region names, which you can get using this call:

```
GET https://cloud.ravellosystems.com/api/v1/elasticIps/locations/
```

Call stays the same -

```
POST https://cloud.ravellosystems.com/api/v1/elasticIps
{ "name": "my_elastic_ip", "description": "This elastic IP is used in my application", "locationType": "REGIONAL", "locationName": "us-central-1" }
```

Regarding the Python SDK, the change will take affect once you update to the latest version - make sure not to install new version before you update your scripts.
def publish_application(self, app, req={"optimizationLevel":"COST_OPTIMIZED"}):

**Comment by ryran** [comment]
Fixed by merging of #68.

Example:

```
$ ravshello -a
Welcome to ravshello, a shell to provision & manage machines with Ravello!
...

/apps> new rsaw-exXXX-B 

Enter a name for your new application [rsaw-exXXX-B]: 

Optionally enter a description for your new app: 

Application 'rsaw-exXXX-B_0' created!

Available publish locations:
  0)  @auto
  1)  us-central1
  2)  Virginia
  3)  Oregon
  4)  us-central-1
  5)  us-east-1
  6)  us-west-1

Select cloud region in which to provision your VMs by entering a number: 2

Ravello now publishing your application (Could take a while)

App auto-stop set for 120 minutes from now

Polling application every 20 secs for next 30 mins to display VM status . . .
Will stop polling when all VMs reach 'STARTED' state
(It won't hurt anything if you cancel status loop early with Ctrl-c)


App VMs in Amazon (Virginia) will auto-stop in 119:38 min at 23:30:38

  server3
     State:              WAITING_TO_START

  server2
     State:              WAITING_TO_START

  server1
     State:              STARTING
```

Of course the `region=` arg for the `/apps new` and `/apps/APPNAME publish` commands works as well. As has always been the case, specifying `region=@auto` sets `optimizationLevel` to "COST_OPTIMIZED".

```
$ ravshello -a /apps new rsaw-exXXX-B name=@auto desc=@auto region=us-east-1
Welcome to ravshello, a shell to provision & manage machines with Ravello!
...

Application 'rsaw-exXXX-B_0' created!

Ravello now publishing your application (Could take a while)

App auto-stop set for 120 minutes from now

Polling application every 20 secs for next 30 mins to display VM status . . .
Will stop polling when all VMs reach 'STARTED' state
(It won't hurt anything if you cancel status loop early with Ctrl-c)


App VMs in Oracle Ravello Cloud (US East 1) will auto-stop in 119:38 min at 23:38:24

  server3
     State:              WAITING_TO_START

  server2
     State:              WAITING_TO_START

  server1
     State:              STARTING
```

**Comment by ryran** [comment]
Went back and found the other places where the old cloud-region info was used and fixed it up.
Note how `query_status` now only shows the region.

```
$ ravshello -aq /apps/rsaw-exXXX-C_0/ query_status
...
App VMs in region us-east-1 will auto-stop in 69:18 min at 01:15:57
```

Same for `ls`.

```
$ ravshello -aq /apps ls depth=1
...
o- apps ........................................ [3 active VMs, 1 of 1 applications published]
  o- rsaw-exXXX-C_0 ................................................... [STARTED in us-east-1]
```
